hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/EchoNoCopyByteArrayProcessor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.cluster.impl;
+
+import com.hazelcast.util.ByteArrayProcessor;
+
+public class EchoNoCopyByteArrayProcessor
```

Comment:
The name of this class doesn't reveal its intent. I'm a but confused about Echo No Copy... but In the code I do see an array copy.